doc/JSON_FLAGS.md

```diff
@@ -883,7 +883,8 @@ Melee flags are fully compatible with tool flags, and vice versa.
 - ```BOMB``` It can be a remote controlled bomb.
 - ```CABLE_SPOOL``` This item is a cable spool and must be processed as such. It has an internal "state" variable which may be in the states "attach_first" or "pay_out_cable" -- in the latter case, set its charges to `max_charges - dist(here, point(vars["source_x"], vars["source_y"]))`. If this results in 0 or a negative number, set its state back to "attach_first".
 - ```CHARGEDIM``` If illuminated, light intensity fades with charge, starting at 20% charge left.
-- ```DIG_TOOL``` If wielded, item is automatically activated when player tries to move onto map tile with ```MINEABLE``` flag.
+- ```DIG_TOOL``` If wielded, digs thorough terrain like rock and walls, as player walks into them.
+- ```DIG_TOOL_POWERED``` If wielded, digs thorough terrain like rock and walls, as player walks into them.  Faster than ```DIG_TOOL``` and uses up power.
```

Comment:
It might be better to split the POWERED flag to be separate. And it already exists, it was merged with #23704 

It is not listed in the JSON_FLAGS.md, which is my fault, I did not realize there is extra doc for existing flags...

Reply:
Not sure, but this is most likely a different thing.

Reply:
I use it to differentiate between tools that have AXE quality but are not powered, like a wood axe, and tools that have AXE quality and are powered, like a chainsaw.

The only difference is that chainsaw has 2 states (or rather 2 items), one for chainsaw turned off, and one for chainsaw turned on. Since jackhammer has only one state - it could be considered as permanently on.

Instead of having "DIG_TOOL_POWERED" flag, you can give the jackhammers 2 flags, that is "DIG_TOOL", like pickaxe has, and "POWERED" on top of it.

In your condition, instead of testing
`u.weapon.has_flag( "DIG_TOOL_POWERED" )`
you would use
`u.weapon.has_flag( "DIG_TOOL" ) && u.weapon.has_flag( "POWERED" )`

Reply:
I agree that it would be a more clean approach to use single `POWERED` flag.